app/services/finance/billing_service.rb

```diff
@@ -30,13 +28,13 @@ def initialize(account_id, options = {})
 
     def call!
       with_lock { call }
-    rescue LockBillingError => error
+    rescue LockBillingError, SpuriousBillingError => error
       report_error(error)
       nil
     end
 
     def call
-      Finance::BillingStrategy.daily(billing_options)
+      Finance::BillingStrategy.daily(validate_options(billing_options))
```

Comment:
I don't know, to me it seems unnecessary to "wrap" it, I would just throw within `def billing_options` directly